### ISSUE TITLE
[BUG] Project Billing - Date is wrong 

### ISSUE THREAD

**Issue by xstable** [eneiluj/cospend-nc#143]
Do the following, like shown at the screen:

1. Set the Date (1) 
2. choose the payer (2)
3. choose the recipient (3)
4. Save  this transaction
5. Then in the list, the entry appears, BUT with a WRONG date (4)

![image](https://user-images.githubusercontent.com/1778068/168418531-9bf87251-8a4d-44da-9c41-bfe6ed0671fb.png)

The date which appear in the list is always today. 
I found that this also happens on regular entries of expenditure, but not yet fully reproduce it.

This may not be noticed, if you enter items up from today in the future, but if you have a list with a long history, and start to enter all the items, this BUG is absolutely reproducible.
This is a big issue, as it's mix up the whole entries you do in the list.

**Comment by eneiluj** [comment]
Hi, thanks for the bug report. I can't reproduce with Cospend v1.4.6 or master branch.

It might just be a display issue in the bill list. I hope the bill dates are correctly set but badly displayed in the bill list.

I need more information:
* What happens if you reload the page? Are the dates still wrong in the bill list? Is the date correct in the bill edition form?
* Which Nextcloud and Cospend version are you using?
* Which language and locale are selected in your Nextcloud user settings?

**Comment by eneiluj** [comment]
Oh ignore my previous comment. You are actually talking about the settlement feature.

It is expected that the settlement bills dates are set to "today". The "Settlement date" field actually lets you select until when you want to consider the bills. It's a way to say "I want everybody's balance to be "0" at this date. But if there are further bills, the real balances won't be "0". Does that make sense to you?

Bottom line is: The auto-settlement feature will always create bills with the current date. You can change it if you want.

It does not seem like a bug to me. The bills could indeed be created on the selected date. Maybe it makes more sense indeed.

**Comment by eneiluj** [comment]
And by the way, individual settlement is not affected by the "Settlement date". It just creates a bill so that the left side member balance becomes "0" by paying something to the right side member. This bill will always be created today.

**Comment by eneiluj** [comment]
Auto-settlement bills are actually already created with the correct date: one minute before the selected "Settlement date".

**Comment by xstable** [comment]
Thanks for your response. 
Ok, I see, then the wording is wrong in German. It's not  Abrechnungsdatum, but "Abrechnungszeitraum bis".

So in my opinion the feature to set a date for the settlement is additional needed.

Why?
1. It's the general case, that people may got the money from the other once somewhere in the last day's and edit the cospend list some days (or weeks) later. 
2. If you start with cosplay, you may want to edit a given range of history for your statistics.

On the other hand, there also where issues with the date for general entries. 
> What happens if you reload the page? Are the dates still wrong in the bill list? Is the date correct in the bill edition form?
Yes, stay's the same on reload, no there is no correct data, even not in the edit dialog.

> Which Nextcloud and Cospend version are you using?
Nextcloud 23.0.4 | Cospay 1.4.6

> Which language and locale are selected in your Nextcloud user settings?
German language
